### ISSUE TITLE
Issue with reactive content

### ISSUE THREAD

**Issue by vouillon** [ocsigen/eliom#220]
We are building a reactive HTML list:
```
       lwt s3 = Eliom_shared.ReactiveData.RList.Lwt.map_p display2 s2 in
       let l = R.ul ?a s3 in
```
where `display2` returns items `Html5.F.li [Html5.D.button ...]`.

Initially, the list is empty. The problem we have is that, when inserting an item, only the `li` element is inserted, with no button inside. I believe what happens is that the item is instantiated twice, so we have two instances of the `li` instance, one that is inserted on the page, and the other one, containing the button, which is discarded. (I have checked that the button has indeed a `li` parent, which is not the `li` element on the page.)

I'm not sure whether this issue comes from a recent change in eliom or in reactiveData.

**Comment by vasilisp** [comment]
Was this working until recently, or it came up in new code? I will take a look ASAP in any case.

**Comment by vasilisp** [comment]
It comes from changes that I have recently made to reactiveData. I apologize.

ocsigen/reactiveData@583fea9f227e46829d42f092cacfa879c88ef2fb fixes the issue on my test environment, but I am not sure this qualifies as anything more than a workaround. I think we need to rework the recently reworked reactiveData internals.

**Comment by vouillon** [comment]
We have a workaround for now: we use a `D.li` element instead of a `F.li` element. So, there is no urgency.

**Comment by vasilisp** [comment]
I am glad, but the underlying issue is quite serious and needs to be addressed soon :).

I think our best option is to [restore mutable references](https://github.com/ocsigen/reactiveData/compare/restore-mutable-ref) in reactiveData. Switching to signals was misguided on my part.

**Comment by vasilisp** [comment]
Fixed via ocsigen/reactiveData@87be26696038da8fca80006971ccec792640e73f .
